File: 백준/Silver/18126. 너구리 구구/너구리 구구.js
const fs = require('fs');
const input = fs.readFileSync(0).toString().split("\n");

const n = Number(input[0]);
const graph = new Array(n+1).fill(0).map(() => []);
const visited = new Array(n+1).fill(-1);

const dfs = (V, prev) => {
  for(let i = 0; i < graph[V].length; i++) {
    if(visited[graph[V][i][0]] === -1) {
      visited[graph[V][i][0]] = prev + graph[V][i][1];
      dfs(graph[V][i][0], visited[graph[V][i][0]]);
    }
  }
} 

for(let i = 1; i <n ; i++) {
  const [a,b,c] = input[i].split(" ").map(Number);
  graph[a].push([b,c]);
  graph[b].push([a,c]);
}


visited[1] = 0;
dfs(1, 0);
console.log(Math.max(...visited));

File: 백준/Silver/18126. 너구리 구구/너구리 구구.py
import sys
sys.setrecursionlimit(10**6)
input = sys.stdin.readline

def solution():
    n = int(input())
    graph = [[] for _ in range(n + 1)]
    visited = [-1] * (n + 1)

    for _ in range(n - 1):
        a, b, c = map(int, input().split())
        graph[a].append((b, c))
        graph[b].append((a, c))

    def dfs(v, prev):
        for nxt, weight in graph[v]:
            if visited[nxt] == -1:
                visited[nxt] = prev + weight
                dfs(nxt, visited[nxt])

    visited[1] = 0
    dfs(1, 0)
    print(max(visited))

solution()